sdks/go/pkg/beam/core/graph/bind.go

```diff
@@ -249,34 +249,37 @@ func tryBindInbound(candidate typex.FullType, args []funcx.FnParam, isMain bool)
 				}
 			}
 
-		case typex.GBKType:
+		case typex.CoGBKType:
 			if args[0].Kind != funcx.FnValue {
-				return nil, kind, fmt.Errorf("Key of %v cannot bind to %v", t, args[0])
+				return nil, kind, fmt.Errorf("key of %v cannot bind to %v", t, args[0])
```

Comment:
<!--thread_id:cc_166095125_t; commit:977a4f133a10d957b90566426581193d93041321; resolved:1-->
<!--section:context-quote-->
> **wcn3** wrote:
> general comment: in these cases, we record why the the thing is not acceptable, but perhaps we could also record what our expectation is. The line number information is not included in these error strings, so it would help making the chain of causality more clear. This has bitten me in the past with coders in particular.
> 
> I note that errors of this requested style are done elsewhere, particularly in the graph package. Should we invest some time to get to a common level of consistency in this space, or at least produce guidance on writing these error messages?

<!--section:body-->
I'm open to that suggestion, although not in this PR.

Reply:
Not intended to be part of this PR. I can open a JIRA to track the request, just wanted to get your take on the idea.